rust/lance-table/src/format/manifest.rs

```diff
@@ -214,9 +214,19 @@ impl Manifest {
     /// Replaces the metadata of the field with the given id with the given key-value pairs.
     ///
     /// If the field does not exist in the schema, this is a no-op.
-    pub fn replace_field_metadata(&mut self, field_id: i32, new_metadata: HashMap<String, String>) {
+    pub fn replace_field_metadata(
+        &mut self,
+        field_id: i32,
+        new_metadata: HashMap<String, String>,
+    ) -> Result<()> {
         if let Some(field) = self.schema.field_by_id_mut(field_id) {
             field.metadata = new_metadata;
+            Ok(())
+        } else {
+            Err(Error::invalid_input(
+                format!("field with id {} for replaceFieldMetadata", field_id),
```

Comment:
the error message seems not clear, can we use "Cannot replace field with id {}: not exist"?

Reply:
The actual message is like :

java.lang.IllegalArgumentException: Invalid user input: field with id 2147483647 for replaceFieldMetadata, /Users/Nathan/workspace/lance/rust/lance-table/src/format/manifest.rs:228:17

Because we used an invaild_input function, the message has been formatted. I can construct a formatted error type for this case if you don't feel heavy?

Reply:
The main reason I was asking that is because it does not say why the field is is invalid, which is because it does not exist in the schema

Reply:
Also I don't quite understand why we want to add "for replaceFieldMetadata" in the message, that is the java method name you are putting in the rust.

Reply:
Yeah....I guessed this was a bad inference by code assistance and I didn't notice